src/index.coffee

```diff
@@ -33,13 +32,30 @@ Method.define read, String, "buffer", readBuffer
 stream = require "stream"
 {promise} = require "when"
 
-Method.define read, stream.Readable, (stream) ->
+# Stringifies a stream's buffer according to the given encoding.
+processBuffer = (stream, encoding = "utf8") ->
   buffer = ""
   promise (resolve, reject) ->
-    stream.on "data", (data) -> buffer += data.toString()
+    stream.on "data", (data) -> buffer += data.toString(encoding)
     stream.on "end", -> resolve buffer
     stream.on "error", (error) -> reject error
 
+# Extracts a stream's raw buffer.
+extractBuffer = (stream) ->
+  buffer = ""
+  promise (resolve, reject) ->
+    stream.on "data", (data) -> buffer += data.toString()
+    stream.on "end", -> resolve new Buffer(buffer)
+    stream.on "error", (error) -> reject error
+
+Method.define read, stream.Readable, processBuffer
+Method.define read, stream.Readable, String, processBuffer
+Method.define read, stream.Readable, undefined, extractBuffer
+Method.define read, stream.Readable, "binary", extractBuffer
+Method.define read, stream.Readable, "buffer", extractBuffer
```

Comment:
This is a good idea, but I don't think this is right. You're converting the buffer to an UTF-8 string before turning it back into a buffer. You definitely should use `Buffer::concat` there. Remember, don't optimize things on a hunch like that. In this case, the problem is even worse, because you have no reason to think that converting to UTF-8 and then back to a Buffer is faster, or even the same thing.

Again, the idea is good—make reading a stream parallel with reading a file.